internal/hcl/evaluator.go

```diff
@@ -762,11 +762,19 @@ func (e *Evaluator) evaluateVariable(b *Block, inputVars map[string]cty.Value) (
 
 	attrType := attributes["type"]
 	if override, exists := inputVars[b.Label()]; exists {
-		return e.convertType(b, override, attrType)
+		val, err := e.convertType(b, override, attrType)
+		if err == nil {
+			return val, nil
+		}
+
+		return override, nil
```

Comment:
when converting a variable input to the "type" definition, failures to convert results in "nil" values returned. Thus it's safer/more complete to default to returning back the non converted variable input with failures.